### PR TITLE
11860 Fix SP filing JSON

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#11860

*Description of changes:*
- removed conditional address validation per 11854
- deleted obsolete "registration NR"
- used "tombstone NR" instead
- updated unit test
- app version = 3.5.15

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Registration/RegAddEditOrgPerson.vue [hunk 1]
```diff
@@ -224,7 +224,7 @@
                 ref="mailingAddressNew"
                 class="mt-4 mb-n6"
                 :editing="true"
-                :schema="enableRules ? PersonAddressSchema : null"
+                :schema="PersonAddressSchema"
```
Comment (severinbeauvais) [inline src/components/Registration/RegAddEditOrgPerson.vue:227]: Same fix as 11854 (which was on a side branch).

According to 11837, we'll review this later.

File: src/interfaces/store-interfaces/state-interfaces/registration-state-interface.ts [hunk 2]
```diff
@@ -7,19 +7,12 @@ export interface RegistrationNameRequestIF {
   nrNumber: string
 }
 
-export const EmptyRegistrationNameRequest: RegistrationNameRequestIF = {
```
Comment (severinbeauvais) [inline src/interfaces/store-interfaces/state-interfaces/registration-state-interface.ts:10]: No longer needed in state (store).

RegistrationNameRequestIF is now only used to declare the filing object sent to the API.

File: src/mixins/filing-template-mixin.ts [hunk 3]
```diff
@@ -344,7 +345,7 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
         business: {
           identifier: this.getTempId,
           naics: this.getRegistration.naics,
-          taxId: this.getRegistration.businessNumber
+          taxId: this.getRegistration.businessNumber || undefined // can't be null
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:348]: Using `undefined` makes this prop NOT part of the resulting JSON.

(This magic is done by JS' internal object <-> JSON conversion.)

File: tests/unit/filing-template-mixin.spec.ts [hunk 4]
```diff
@@ -82,13 +80,20 @@ describe('Registration Filing', () => {
...
 
-    store.state.stateModel.registration.nameRequest = {
-      legalName: 'My Approved Name',
-      legalType: 'SP',
-      nrNumber: 'NR 1234567'
+    store.state.stateModel.nameRequest = {
+      entityType: 'SP',
+      nrNumber: 'NR 1234567',
+      details: { approvedName: 'My Approved Name' }
     }
```
Comment (severinbeauvais) [inline tests/unit/filing-template-mixin.spec.ts:87]: Now using "tombstone NR".

File: tests/unit/test-data/registration.json [hunk 5]
```diff
@@ -40,7 +40,7 @@
       },
       "businessType": "SP",
       "contactPoint": {
-        "email": "eleven@email.com",
+        "email": "eleven@example.com",
```
Comment (severinbeauvais) [inline tests/unit/test-data/registration.json:43]: This is a non-routable email address, whereas `email.com` is a legitimate domain.

**Review by Lcyc29 — APPROVED** [review]
Everything looks good. Thanks for making this big fix.

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by cameron-freshworks — APPROVED** [review]

**Pull request merged**